ansible/roles/post-recheck-comment/tasks/main.yaml

```diff
@@ -0,0 +1,64 @@
+---
+
+- name: Store Verified-1 regex pattern
+  set_fact:
+    failure_pattern: '^Patch Set {{ patchset }}: Verified-1'
+
+- name: Wait for Verified-1 comment from {{ ci_user }}
+  register: comments
+  raw: gerrit query change:{{ change_id }} --comments --format JSON
+  retries: 6
+  delay: 10
+  until: >
+    comments.rc == 0 and
+    (comments.stdout_lines[0] | from_json).comments
+    | selectattr('reviewer.username', 'equalto', ci_user)
+    | map(attribute='message')
+    | list
```

Comment:
We think that we can improve readability of this conditional (and any other expression which includes comments) by employing a custom filter. Let's say this filter would be named `extract_messages`. `extract_message` would do the same as the following expression:

```
(comments.stdout_lines[0] | from_json).comments
| selectattr('reviewer.username', 'equalto', ci_user)
| map(attribute='message')
```

Play would then look like this:

```
 name: Wait for Verified-1 comment from {{ ci_user }}
 register: comments
 raw: gerrit query change:{{ change_id }} --comments --format JSON
 retries: 6
 delay: 10
 until: >
   comments.rc == 0 and
   comments | extract_messages | last | regex_search(failure_pattern)
```

Here you can find an example of a custom filter which we tested: https://gist.github.com/sodar/a530ad10cc81cbe47a494f398b0bfd31

What do you think about this approach?

Reply:
Good idea! I thought about using a custom filter but it seems I didn't google hard enough. The only small disadvantage is requiring user to jump between files and that it's not really obvious where `extract_messages` comes from (imagine reading this code and furiously trying to google `ansible extract_messages`). But with a comment, I think it would be ok.

Reply:
Yes, jumping between the files could cause problems, but in the end this conditional will be more understandable.

As to the naming, maybe we should namespace it? E.g. `cw_extract_messages` - contrail windows extract messages. Still, someone might google it 😄. Your call.

Also, IIRC `filter_plugins` directory can be either in root directory or in the role directory. Since it is used only here, maybe it should be in role directory.